File: Squared tiles.py
mat=[
[ '*','*','*','*','*','*'],
[ '*','*','*','*','*','*'],
[ '*','*','*','*','*','*'],
[ '*','*','*','*','*','*'],
[ '*','*','*','*','*','*'],
[ '*','*','*','*','*','*']
]


def place_matrix2(user):
    flag=0
    for i in range(6):
        for j in range(6):
            if validate(i,j,user):
                place_matrix(i,j,user)
                flag=1
                break
        if flag==1:
            break

def print_matrix():
    for i in range(6):
        for j in range(6):
            print(mat[i][j],end=" ")
        print()

def place_matrix(x,y,z):
    for i in range(x,x+z):
        for j in range(y,y+z):
            mat[i][j]="1"

def validate(x,y,z):
    if (x+z-1)>5 or (y+z-1)>5:
        return False
    for i in range(x,x+z):
        for j in range(y,y+z):
            if mat[i][j]=="1":
                return False
    return True
def computer_check(user,level):
    if level==1 and user==3:
        place_matrix2(3)
    elif level==1 and user==2:
        place_matrix2(3)
    #elif level==2:
        #place_matrix2(user)
    elif level==3:
        place_matrix2(2)
    else:
        place_matrix2(2)
def assign_mat(mm):
    prev = [
        ['*', '*', '*', '*', '*', '*'],
        ['*', '*', '*', '*', '*', '*'],
        ['*', '*', '*', '*', '*', '*'],
        ['*', '*', '*', '*', '*', '*'],
        ['*', '*', '*', '*', '*', '*'],
        ['*', '*', '*', '*', '*', '*']
    ]
    for i in range(6):
        for j in range(6):
            prev[i][j]=mm[i][j]
    return prev


level=1
n=0
while(1):
    flag = 0
    user=int(input("Enter your squared matrix(2*2 or 3*3): "))
    for i in range(6):
        for j in range(6):
            if validate(i,j,user):
                place_matrix(i,j,user)
                flag=1
                break
        if flag==1:
            n=1
            break
    print()
    print()
    if flag==0:
        print("AI Win!!!!!")
        break
    print("user: ")
    print_matrix()
    print("Computer: ")
    computer_check(user,level)
    print_matrix()
    level+=1
if n==0:
    print("User win")

File: squared tiles.cpp
#include<bits/stdc++.h>
using namespace std;
string mat[6][6];
int first;

int place_matrix3(int x,int y,int z)
{
    for (int i=x; i<(x+z); i++)
    {
        for (int j=y; j<(y+z); j++)
        {
            mat[i][j]="0";

        }

    }


}
bool validate(int x,int y,int z)
{
    if ((x+z-1)>5 || (y+z-1)>5)
    {
        return false;


    }

    for (int i=x; i<(x+z); i++)
    {
        for(int j=y; j<(y+z); j++)
        {
            if (mat[i][j]=="1" || mat[i][j]=="0")
                return false;

        }

    }

    return true;

}
int print_matrix()
{
    for (int i=0; i<6; i++)
    {
        for (int j=0; j<6; j++)
        {
            cout<<mat[i][j]<<" ";

        }
        cout<<endl;

    }


}


int place_matrix2(int user)
{
    int flag=0;
    int ch=0;
    for (int i=0; i<6; i++)
    {
        for (int j=0; j<6; j++)
        {
            if (validate(i,j,user))
            {
                place_matrix3(i,j,user);
                flag=1;
                ch=1;
                break;
            }

        }
        if (flag==1)
            break;


    }
    return ch;



}


int place_matrix(int x,int y,int z)
{
    for (int i=x; i<(x+z); i++)
    {
        for (int j=y; j<(y+z); j++)
        {
            mat[i][j]="1";

        }

    }
}




int computer_check(int user,int level)
{
    int v;

    if (level==1 && user==3)
    {
        v=place_matrix2(3);
    }

    else if (level==1 && user==2)
    {
        v=place_matrix2(3);
    }

    else if (level==2 && user == 2 and first==2)
    {
        v=place_matrix2(2);
    }

    else if (level==2 && user == 2)
    {
        v=place_matrix2(3);
    }

    else if (level==2 && user == 3)
    {
        v=place_matrix2(3);

    }

    else if (level==3)
    {
        v=place_matrix2(2);
    }

    else
    {
        v=place_matrix2(2);
    }

    return v;



}





int main()
{
    int label=1;

    for(int i=0; i<6; i++)
    {
        for(int j=0; j<6; j++)
        {
            mat[i][j]="*";
        }
    }
    while(1)
    {
        int flag=0;
        int check=0;
        int user;

        while(1)
        {
            printf("user input(2*2 and 3*3): ");
            cin>>user;
            if(user==2 || user==3)
            {
                break;
            }
        }


        if (label==1)
        {
            first=user;
        }

        for (int i=0; i<6; i++)
        {
            for (int j=0; j<6; j++)
            {
                if (validate(i,j,user))
                {
                    place_matrix(i, j, user);
                    printf("User:\n");
                    print_matrix();
                    check=1;
                    flag=1;
                    break;
                }

            }
            if (flag==1)
            {
                break;
            }

        }
        if (check == 0)
        {
            printf("AI win !!!!\n");
            break ;
        }
        printf("AI:\n");
        int v=computer_check(user, label);
        label = label + 1;
        if (v==0)
        {
            printf("user win !!!!");
            break;
        }
        print_matrix();


    }

}
